### ISSUE TITLE
[C++ API] tf.reset_default_graph() equivalent

### ISSUE THREAD

**Issue by Pipoderoso** [tensorflow/tensorflow#20058]
Does the C++ API has an equivalent method to `tf.reset_default_graph()` ?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by asimshankar** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

If you think we've misinterpreted a bug, please comment again with a clear explanation, as well as all of the information requested in the [issue template](https://github.com/tensorflow/tensorflow/issues/new). Thanks!

That said, in the C++ API the graph is always explicit - the `Graph` object or some representative of it is always provided to the API functions.

In other words, there is no notion of a "default graph", so there is no `tf.reset_default_graph()`.
Hope that helps.
